crates/common/src/compile.rs

```diff
@@ -322,9 +378,14 @@ impl Display for SizeReport {
     }
 }
 
-/// Returns the size of the deployed contract
-pub fn deployed_contract_size<T: Artifact>(artifact: &T) -> Option<usize> {
-    let bytecode = artifact.get_deployed_bytecode_object()?;
+/// Returns the deployed or init size of the contract.
+fn contract_size<T: Artifact>(artifact: &T, initcode: bool) -> Option<usize> {
+    let bytecode = if initcode {
+        artifact.get_bytecode_object()?
```

Comment:
The initcode check should probably also include the length of constructor arguments, since those become part of the initcode. But for dynamic length types we will have to make an assumption on length. One option is assume zero length dynamic types and let the margin column implicitly document how long dynamic type args can be. Another option is to continue excluding constructor arg length from the initcode length and document this limitation

Reply:
I'm thinking we can continue excluding constructor arg length and document it in the `--help` output for `--sizes` like
```
    /// Print compiled contract sizes.
   /// Constructor argument length is not included in the calculation of initcode size.
    #[arg(long)]
    #[serde(skip)]
    pub sizes: bool,
```
does this sound good? @mds1

Reply:
@zerosnacks does the above look good to add it in this PR?

Reply:
> I'm thinking we can continue excluding constructor arg length and document it in the `--help` output for `--sizes` 

This is ok with me as long as @zerosnacks agrees :)